src/lib.rs

```diff
@@ -16,13 +16,13 @@ pub type Data<T> = <T as GraphObject>::Data;
 /// Abstract object in a graph, eg. node or edge.
 pub trait GraphObject {
     /// Identifier of all graph objects.
-    type Id: Clone;
+    type Id;
```

Comment:
Id's that can't be cloned are pretty annoying to work with imo, means that you're forever holding references to the container, and you can't mutate the cointainer while holding those references.

Reply:
This is a choice I think. You can either make `Id` less polymorphic here _or_ require your Ids to be clone-able when writing functions, like:

```rust
fn foo<G>(graph: G) 
  where
    <G::Node as GraphObject>::Id: Clone
{ ... }
```

In `osrank` we chose this approach and it's working out well so far. Do you foresee we can do the same here?

Reply:
Yeah makes sense.